node_modules/col-whiteboards/lib/api.js

```diff
@@ -0,0 +1,771 @@
+  var options = {
+    'where': {
+      'socket_id': socketId
+    }
+  };
+
+  DB.WhiteboardSession.find(options).complete(function(err, whiteboardSession) {
+    if (err) {
+      log.error({'err': err, 'id': id, 'socketId': socketId}, 'Failed to retrieve the whiteboard session to delete');
+      return callback({'code': 500, 'msg': err.message});
+    } else if (!whiteboardSession) {
+      log.debug({'err': err, 'id': id, 'socketId': socketId}, 'The whiteboard session that is being deleted could not be found');
+      return callback({'code': 404, 'msg': 'The whiteboard session that is being deleted could not be found'});
+    }
+
+    // Delete the whiteboard session from the DB
+    whiteboardSession.destroy().complete(function(err) {
+      if (err) {
+        log.error({'err': err, 'socketId': socketId}, 'Failed to delete a whiteboard session');
+        return callback({'code': 500, 'msg': err.message});
+      }
+
+      // Send the updated list of online users to the users in the whiteboard
+      notifyOnlineUsers(whiteboardSession.whiteboard_id);
+
+      return callback();
+    });
+  });
+};
+
+/**
+ * Delete all whiteboard sessions. This is necessary when the server restarts, as all
+ * websockets will have disconnect without the disconnect logic being executed
+ *
+ * @param  {Function}       [callback]                      Standard callback function
+ * @param  {Object}         [callback.err]                  An error that occurred, if any
+ * @api private
+ */
+var deleteWhiteboardSessions = function(callback) {
+  callback = callback || function() {};
+
+  var options = {
+    'truncate': true
+  };
+
+  DB.WhiteboardSession.destroy(options).complete(function(err) {
+    if (err) {
+      log.error({'err': err}, 'Failed to delete all whiteboard session');
+      return callback({'code': 500, 'msg': err.message});
+    }
+
+    return callback();
+  });
+};
+
+// Delete all whiteboard sessions when the server starts
+deleteWhiteboardSessions();
+
+/* WHITEBOARD ELEMENTS */
+
+/**
+ * Store a new element was added to the whiteboard canvas and inform
+ * the online users in the whiteboard of the new element
+ *
+ * @param  {Socket}         socket                          The socket on which the new whiteboard element was sent
+ * @param  {Object}         element                         The deserialized whiteboard element that was added
+ * @param  {Function}       [callback]                      Standard callback function
+ * @param  {Object}         [callback.err]                  An error that occurred, if any
+ * @api private
+ */
+var addWhiteboardElement = function(socket, element, callback) {
+  callback = callback || function() {};
+
+  // Notify all online users in the whiteboard that a new whiteboard element was added.
+  // This will not send a notification over the socket on which the new element was sent
+  socket.broadcast.to(socket.whiteboard.id).emit('addElement', element);
```

Comment:
Should this broadcast whappen when `upsertWhiteboardElement`'s callback function got triggered?